tallycash/extension
# Decoding EIP2612 Permit Messages - 🦄  Uniswap and beyond...

This PR Adds a formatted screen for signing uniswap permit transactions.

Closes #1100

### To test:
- Navigate to https://app.uniswap.org/#/swap?chain=mainnet
- [ ] Attempt to permit an asset you have not previously granted permission for.
- [ ] Attempt to permit an asset you have previously granted permission for.
- [ ] Attempt to permit an asset you have not previously granted permission for and is not in our current token list.
- [ ] Attempt to permit an asset and sign message with Ledger (I have not been able to test this yet)

You should see a beautifully formatted message pop up!

Signing typed data for Dapps other than Uniswap should continue to work as usual - with the added benefit that addresses will attempt to display their ens names and have links to Etherscan.


<img width="380" alt="Screen Shot 2022-03-10 at 3 16 27 PM" src="https://user-images.githubusercontent.com/94649004/157765040-fbfcc2b8-3ac7-4adc-9d48-a6c95b1636ab.png">
